src/language/class-modifiers.md

```diff
@@ -0,0 +1,240 @@
+---
+title: Class modifiers
+description: Modifier keywords for class declarations to control external library access.
+---
+
+Class modifiers control how a class or a mixin can be implemented, extended,
+or mixed in. 
+
+The full set of modifiers that can appear before a class declaration are `base`,
+`final`, `interface`, `sealed`, `abstract`, and `mixin`. 
+Only the `base` modifier can appear before a mixin declaration. The modifiers do
+not apply to other declarations like `enum`, `typedef`, or `extension`.
+
+When deciding whether or not to use class modifiers, consider the following questions:
+1. Is [any access](#no-modifier) fine?
+2. If not, what do you want to restrict?
+    - [Implementation](#base)?
+    - [Inheritance](#interface)?
+    - [Both](#final)?
+    - [Subtyping](#sealed) by outside libraries at all?
+3. Then, does it need to be [abstract](#abstract)?
+4. And lastly, should it be a [class, mixin, or mixin class](#class-mixin-or-mixin-class)?
+
+## No modifier
+
+Use a `class` or `mixin` declaration without a modifier to **allow unrestricted
+access** to outside libraries. The standard behaviors apply:
+
+- [Construct][] new instances of a class
+- [Extend][] a class to create a new subtype
+- [Implement][] a class's interface
+- [Mix in][] a mixin or mixin class
+
+## `base`
+
+Use the `base` modifier to **disallow implementation** of a class or mixin.
```

Comment:
I'd focus on the other direction: It *enforces inheritance* of the class or mixin's implementation.

Preventing implementation is a means toward that goal, not a goal in itself.
It's probably mainly a philosophical distinction, but it underlines *why* you'd add `base`. It's not to prevent implementation. If there were other ways to create a subclass which doesn't inherit the superclass implementation, preventing implementation wouldn't give you anything useful by itself.
There isn't, which is why `base` only prevents implementation. That's enough to reach its goal of ensuring that all subclasses inherit the implementation.